### PR TITLE
Set SemverCompatible to false for Conan packages

### LINKED ISSUE #21250: Error 500 if upload non SemVers Conan packages
### Description

We are trying Gitea as repository.

We are experimenting the same issue that this case using Conan package:

https://github.com/go-gitea/gitea/issues/20603#issue-1324905900

[****](https://github.com/go-gitea/gitea/issues/20603#issuecomment-1247069270)

If we upload a conan package using nonSem version, we have error 500 on repo package pages.

To fix this issue, I update semver_compatible from true to false :

```
giteadb=# UPDATE package SET semver_compatible = 'f' WHERE id = 7;
UPDATE 1
giteadb=# UPDATE package SET semver_compatible = 'f' WHERE id = 9;
UPDATE 1
giteadb=# select * from package;
 id | owner_id | repo_id |  type   |     name     |  lower_name  | semver_compatible
----+----------+---------+---------+--------------+--------------+-------------------
  1 |        2 |       0 | generic | cirros       | cirros       | f
  2 |        4 |       0 | conan   | boost        | boost        | t
  4 |       12 |       0 | conan   | b2           | b2           | t
  5 |       12 |       0 | conan   | boost        | boost        | t
  6 |       12 |       0 | conan   | grpc         | grpc         | t
 10 |       12 |       0 | conan   | openssl      | openssl      | t
 11 |       12 |       0 | conan   | zlib         | zlib         | t
  7 |       12 |       0 | conan   | libbacktrace | libbacktrace | f
  9 |       12 |       0 | conan   | opengv       | opengv       | f
(9 rows)

giteadb=#
```

There is no command on conan upload cli to fix semvers to true or false.

Is this a bug in Gitea ?

Thanks,

### Gitea Version

1.18-dev

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

Ubuntu 22

### How are you running Gitea?

Install gitea on vmware virtual machine, ubuntu22 using the binary.

### Database

PostgreSQL

### LINKED ISSUE #21284: Conan package malformed version
### Description

After uploading the libbacktrace/cci.20210118 conan package I got internal server error (500) when was trying to see packages in web interface.

List of uploaded packages:

```
conan search -r ve-infra
Existing package recipes:

boost/1.77.0
bzip2/1.0.8
flatbuffers/2.0.5
libbacktrace/cci.20210118
openssl/1.0.2u
sqlite3/3.38.1
yaml-cpp/0.7.0
zlib/1.2.12

```

Gitea log records:
```
2022/09/28 11:12:28 ...kages/conan/conan.go:62:apiError() [E] [633401ec-13] Malformed version: cci.20210118
2022/09/28 11:13:28 ...kages/conan/conan.go:62:apiError() [E] [633401ec-14] unexpected EOF
2022/09/28 11:14:28 ...kages/conan/conan.go:62:apiError() [E] [63340228] unexpected EOF
2022/09/28 11:14:40 .../web/user/package.go:58:ListPackages() [E] [63340270] GetPackageDescriptors: Malformed version: cci.20210118
2022/09/28 11:14:48 .../web/user/package.go:58:ListPackages() [E] [63340278-8] GetPackageDescriptors: Malformed version: cci.20210118
2022/09/28 11:14:50 .../web/user/package.go:58:ListPackages() [E] [6334027a] GetPackageDescriptors: Malformed version: cci.20210118
2022/09/28 11:14:50 .../web/user/package.go:58:ListPackages() [E] [6334027a-2] GetPackageDescriptors: Malformed version: cci.20210118
2022/09/28 11:15:06 .../web/user/package.go:58:ListPackages() [E] [6334028a] GetPackageDescriptors: Malformed version: cci.20210118
2022/09/28 11:15:54 .../web/user/package.go:58:ListPackages() [E] [633402ba] GetPackageDescriptors: Malformed version: cci.20210118
2022/09/28 11:18:56 .../web/user/package.go:58:ListPackages() [E] [63340370] GetPackageDescriptors: Malformed version: cci.20210118
```

I'm unable to remove the libbacktrace/cci.20210118 package, got the same error:

```
2022/09/28 11:34:35 ...kages/conan/conan.go:62:apiError() [E] [6334071b-2] Malformed version: cci.20210118
2022/09/28 11:34:35 ...kages/conan/conan.go:62:apiError() [E] [6334071b-3] Malformed version: cci.20210118
2022/09/28 11:34:35 ...kages/conan/conan.go:62:apiError() [E] [6334071b-4] Malformed version: cci.20210118
```

### Gitea Version

1.17.2

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

I have downloaded binary gitea from official site.
It runs on host by systemd service.

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by KN4CK3R**
Fixes #21250
Related #20414

Conan packages don't have to follow SemVer.
The migration fixes the setting for all existing Conan and Generic (#20414) packages.

### CODE REVIEW

**Comment by wxiaoguang** [comment]
I have a question, is it necessary to report errors and show 500 when the package version can not be parsed for SemverCompatible=true? 500 error seems not friendly to user, and they can not even delete the package.

**Comment by KN4CK3R** [comment]
If SemverCompatible=true there is a check in the upload process to enforce a valid version. This check was missing for Conan packages because Semver is not enforced by the spec. In this case a 500 error is correct because an invalid version should not occur.

**Review by wxiaoguang — APPROVED** [review]

File: models/migrations/migrations.go [hunk 1]
```diff
@@ -413,6 +413,8 @@ var migrations = []Migration{
 	NewMigration("Add badges to users", createUserBadgesTable),
 	// v225 -> v226
 	NewMigration("Alter gpg_key/public_key content TEXT fields to MEDIUMTEXT", alterPublicGPGKeyContentFieldsToMediumText),
+	// v226 -> v227
+	NewMigration("Fix package semver field", fixPackageSemverField),
```
Comment (delvh) [inline models/migrations/migrations.go]: I think a notice that this affects "conan" and "generic" packages specifically would be a good idea.
```suggestion
	NewMigration("Conan and generic packages do not need to be semantically versioned", fixPackageSemverField),
```

**Review by delvh — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

**Comment by wxiaoguang** [comment]
Merged, please send backport ~

**Pull request merged**